### PR TITLE
Refactor/activity class 

### PR DESCRIPTION
**Pull request by m1073496**
## What does this PR do?

- Refactors Activity class and test
- Reorganizes these files for readability
- Makes methods and tests dynamic
- Adds data to fake data files needed for testing methods that look at a week's worth of data
- Comments left for methods that work but are unused in UI

## How should it be tested in the UI?

- No UI application yet, can be tested in the terminal

### CODE REVIEW

File: src/data/fakeData.js [hunk 1]
```diff
@@ -204,6 +204,48 @@ export const fakeActivityData = [{
...
+    "minutesActive": 116,
+    "flightsOfStairs": 200
+  },
+  {
+    "userID": 3,
+    "date": "2019/06/14",
+    "numSteps": 7402,
+    "minutesActive": 116,
+    "flightsOfStairs": 33
+  },
```
Comment (JeffShepherd) [inline src/data/fakeData.js:248]: woowwweee

File: test/activity-test.js [hunk 2]
```diff
@@ -1,306 +1,112 @@
...
-      dailyStepGoal: 9000,
-      friends: [1, 3, 4]
-    });
+  beforeEach(function () {
+    userRepo = new UserRepo(fakeUserData);
+    userOne = userRepo.users[0];
+    userTwo = userRepo.users[1];
+    userThree = userRepo.users[2];
+    activityRepo = new Activity(fakeActivityData);
+  });
```
Comment (JeffShepherd) [inline test/activity-test.js:23]: Solid way to DRY this up significantly

File: src/Activity.js [hunk 3]
```diff
@@ -1,44 +1,77 @@
...
-      return acc += elem.minutesActive;
-    }, 0) / 7).toFixed(1));
-  }
-  accomplishStepGoal(id, date, userRepo) {
-    let userStepsByDate = this.activityData.find(data => id === data.userID && date === data.date);
-    if (userStepsByDate.numSteps === userRepo.dailyStepGoal) {
+    this.activityData = activityData;
+  };
+
+  calculateMilesWalked(user, date, userRepo) {
```
Comment (JeffShepherd) [inline src/Activity.js]: userRepo param not used in this function. Should this be removed?

File: src/Activity.js [hunk 4]
```diff
@@ -1,44 +1,77 @@
...
+  };
+
+  findMinutesActiveAverage(user, date, userRepo) {
+    let foundWeek = userRepo.findWeekOfData(date, user.id, this.activityData);
+    return (foundWeek.reduce((totalMinutes, day) => {
+      return totalMinutes + day.minutesActive
+    }, 0)) / 7;
+  };
+
+  accomplishStepGoal(user, date, userRepo) {
```
Comment (JeffShepherd) [inline src/Activity.js]: userRepo param not used in this function. Should this be removed?

File: src/data/fakeData.js (hunk 1 above)
Comment (m1073496) [reply src/data/fakeData.js:248]: I had to add a week's worth of activities for one user in order to test methods that use a week's activities 😅

**Comment by Shakikka** [comment]
[Refactor Activity.test](https://trello.com/c/EJlM1MOG/3-refactor-activitytest)

**Comment by Shakikka** [comment]
[Refactor Activity Class](https://trello.com/c/ULofR7Fd/9-refactor-activity-class)

File: src/Activity.js [hunk 5]
```diff
@@ -1,44 +1,77 @@
...
+  };
+
+  getUserTotalsForWeek(user, date, userRepo, stat) {
+    let userWeek = userRepo.findWeekOfData(date, user.id, this.activityData);
+    let statTotal = userWeek.reduce((accu, current) => {
+      accu = accu + current[stat];
+      return accu;
+    }, 0);
+    return statTotal;
+  };
```
Comment (JeffShepherd) [inline src/Activity.js:42]: I like using "accum" and "current" with reduce, but may be best to make them more descriptive.

**Comment by JeffShepherd** [comment]
Unused params removed. Looks good

**Pull request merged**